### PR TITLE
Prevent uncaught rejections on websocket timeout

### PR DESCRIPTION
**Pull request by robgonnella**

### CODE REVIEW

**Comment by robgonnella** [comment]
Fixes: https://github.com/sipwise/janus-client/issues/17

File: test/client-spec.js [hunk 1]
```diff
@@ -15,18 +15,25 @@ var request = {
...
-        }).catch(()=>{
-            done(err);
+        brokenJanusServerMock = new JanusServerMock({
+            port: brokenMockServerPort,
+            triggerHandshakeTimeout: true
         });
+        Promise.all([
+            janusServerMock.init(),
+            brokenJanusServerMock.init()
+        ]).then(() => done()).catch((e) => done(e));
```
Comment (robgonnella) [inline test/client-spec.js]: Just realized I forgot to add the call to `brokenJanusServerMock.close()` in the after hook. Will update.

File: test/client-spec.js [hunk 2]
```diff
@@ -103,6 +110,44 @@ describe('Client', function() {
...
+                process.off('uncaughtException', onUncaughtException);
+                done();
+            }
+
+        });
+
+        process.on('uncaughtException', onUncaughtException)
+
+        client.connect();
+    });
```
Comment (robgonnella) [inline test/client-spec.js]: I'm going to split this test into two tests that cover when the handshakeTimeout occurs before and after the connectionTimeout.

**Comment by hpherzog** [comment]
Thank you for you contribution. We really appreciate that. The change is under review.

**Pull request closed without merging**